doc/tools/coqrst/coqdomain.py

```diff
@@ -455,7 +473,7 @@ class ProductionObject(CoqObject):
 
     # handle_signature is called for each line of input in the prodn::
     # 'signatures' accumulates them in order to combine the lines into a single table:
-    signatures = None
+    signatures = None # FIXME this should be in init, shouldn't it?
```

Comment:
Could be.  The current code works because there's only one active ProductionObject at a time.

The __init__ would be a mouthful:

```
    def __init__(self, name, arguments, options, content, lineno,
                 content_offset, block_text, state, state_machine):
         : 
```

Reply:
It doesn't need to be fixed in this PR, does it?

Reply:
Nope.

Reply:
No, just `__init__(*args, **kwargs)`, I think

Reply:
Yeah.  I've done too much Java.